src/Mpociot/Versionable/Version.php

```diff
@@ -104,4 +104,33 @@ public function diff(Version $againstVersion = null)
         return $diffArray;
     }
 
+    /**
+     * Recursive version of array_diff_assoc.
+     *
+     * @param $array1
+     * @param $array2
+     *
+     * @return array
+     */
+    public function arrayDiffAssocRecursive($array1, $array2)
```

Comment:
Hey @jez500 - have you tested this for performance changes? Is this slower, the same, faster?

Also, is this adding recursivity, which this didn't have before?

Reply:
Hi @nonoesp

This will certainly be slower than the native php functions, but it is also doing more and doesn't cause a fatal error if a property is not a string (eg json fields). That said I have noticed no notable difference in performance with my testing.

Recursively is likely unavoidable with nested arrays, but will only kick in if there are actually nested arrays.